src/System Application/App/Password/src/PasswordDialogImpl.Codeunit.al

```diff
@@ -92,5 +95,21 @@ codeunit 9811 "Password Dialog Impl."
                 exit(false);
         exit(true);
     end;
+
+    [NonDebuggable]
+    procedure ValidateOldPasswordMatch(OldPasswordToCompare: SecretText; OldPasswordEntered: SecretText)
+    begin
+        if not OldPasswordToCompare.IsEmpty() then
+            if OldPasswordToCompare.Unwrap() <> OldPasswordEntered.Unwrap() then
+                Error(OldPasswordMismatchErr);
+    end;
+
+    [NonDebuggable]
+    procedure ValidateNewPasswordUniqueness(OldPasswordToCompare: SecretText; NewPassword: SecretText)
+    begin
+        if not OldPasswordToCompare.IsEmpty() then
```

Comment:
You could reduce nesting here by by doing an "if isempty then exit". Just a suggestion.

Reply:
I agree. I often write like that myself, don't know why I haven't done the same here :)